core/src/main/scripts/importer/cbioportalImporter.py

```diff
@@ -183,102 +201,170 @@ def process_directory(jvm_args, study_directory):
+                (meta_filename, os.path.join(study_directory, meta_dictionary['data_filename'])))
+
+        # Check for GSVA p-values
         elif meta_file_type == MetaFileTypes.GSVA_PVALUES:
             gsva_pvalue_filepair = (
-                (f, os.path.join(study_directory, metadata['data_filename'])))
+                (meta_filename, os.path.join(study_directory, meta_dictionary['data_filename'])))
+
+        # Check for fusion data
+        elif meta_file_type == MetaFileTypes.FUSION:
+            fusion_filepair = (
+                (meta_filename, os.path.join(study_directory, meta_dictionary['data_filename'])))
+
+        # Add all other types of data
         else:
             regular_filepairs.append(
-                (f, os.path.join(study_directory, metadata['data_filename'])))
+                (meta_filename, os.path.join(study_directory, meta_dictionary['data_filename'])))
 
     # First, import cancer types
     for meta_filename, data_filename in cancer_type_filepairs:
         import_cancer_type(jvm_args, data_filename)
 
     # Then define the study
-    if study_metafile is None:
+    if study_meta_filename is None:
         raise RuntimeError('No meta_study file found')
     else:
         # First remove study if exists
-        remove_study(jvm_args, study_metafile)
-        import_study(jvm_args, study_metafile)
+        remove_study(jvm_args, study_meta_filename)
+        import_study(jvm_args, study_meta_filename)
 
     # Next, we need to import sample definitions
     if sample_attr_filepair is None:
         raise RuntimeError('No sample attribute file found')
     else:
         meta_filename, data_filename = sample_attr_filepair
-        import_study_data(jvm_args, meta_filename, data_filename)
+        import_study_data(jvm_args, meta_filename, data_filename, study_meta_dictionary[meta_filename])
 
-    # Now, import everything else except gene panel, gsva & z-score expression
-    # These data types have to be imported last, because they both refer to expression by
-    # source_stable_id. If in the future more types refer to each other, (like
-    # in a tree structure) this could be programmed in a recursive fashion.
+    # Next, import everything else except gene panel, GSVA & Z-score expression
+    # GSVA & Z-score data types have to be imported last, because they both
+    # refer to expression by source_stable_id. If in the future more types refer
+    # to each other, (like in a tree structure) this could be programmed in a
+    # recursive fashion.
     for meta_filename, data_filename in regular_filepairs:
-        import_study_data(jvm_args, meta_filename, data_filename)
+        import_study_data(jvm_args, meta_filename, data_filename, study_meta_dictionary[meta_filename])
+
+    # Import fusion data (after mutation)
+    if fusion_filepair is not None:
+
+        # Replace this when fusion data can be displayed from a fusion
```

Comment:
clearly label this as hack/workaround and add TODO explaining what needs to be done so that this workaround can be removed in the future.